package-puzzles

```diff
@@ -0,0 +1,124 @@
+        
+        # build category seed
+        category_seed = hashlib.new('sha1')
+        category_seed.update(categoryname.encode('utf-8'))
+        category_seed.update(args.seed.encode('utf-8'))
+        category_seed = category_seed.hexdigest()
+
+        # create zipfile
+        zipfilename = os.path.join(args.outdir, "%s.zip" % categoryname)
+        if os.path.exists(zipfilename):
+    # assume blank state for now
+            pass
+    #        # open and gather some state
+    #        zf = zipfile.ZipFile(zipfilename, 'a')
+        else:
+            # create and read in state
+            zf = zipfile.ZipFile(zipfilename, 'w')
+
+        # read in category details (will be pflarr in future)
+        for categorypath in glob.glob(os.path.join(categorydir, "*", "puzzle.moth")):
+            points = categorypath.split(os.sep)[-2] # directory before '/puzzle.moth'
+            categorypath = os.path.dirname(categorypath)
+            print(categorypath)
+            try:
+                points = int(points)
+            except:
+                print("Failed to identify points on: %s" % categorypath, file=sys.stderr)
+                continue
+            puzzle = puzzles.Puzzle(category_seed, categorypath)
+            puzzles_dict[points] = puzzle
+
+        # build mapping, answers, and summary
+        mapping = {}
+        answers = {}
+        summary = {}
+        for points in sorted(puzzles_dict):
+            puzzle = puzzles_dict[points]
+            hashmap = hashlib.sha1(args.seed.encode('utf-8'))
+            hashmap.update(categoryname.encode('utf-8'))
+            hashmap.update(str(points).encode('utf-8'))
+            mapping[points] = hashmap.hexdigest()
+            answers[points] = puzzle['answer']
+            if len(puzzle['summary']) > 0:
+                summary[points] = puzzle['summary']
+
+        # write mapping, answers, and summary
+        write_kv_pairs(zf, os.path.join(categoryname, 'map.txt'), mapping)
+        write_kv_pairs(zf, os.path.join(categoryname, 'answers.txt'), answers)
+        write_kv_pairs(zf, os.path.join(categoryname, 'summary.txt'), summary)
+
+        # write out puzzles
+        for points in sorted(puzzles_dict):
+            puzzle = puzzles_dict[points]
+            puzzlebody = [json.dumps(puzzle.publish())]
+            puzzledir = os.path.join(categoryname, 'content', mapping[points])
+            # build/write json
+
+            # write associated files
+            assoc_files = []
+            for fobj in puzzle['files']:
+                if fobj.visible == True:
+                    assoc_files.append(fobj.name)
+                zf.writestr(os.path.join(puzzledir, fobj.name), \
+                                    fobj.handle.read())
+
+
+            if len(assoc_files) > 0:
+                puzzlebody.append("")
+                puzzlebody.append("## Associated Files:")
+                for f in assoc_files:
+                    puzzlebody.append(f)
```

Comment:
Just want to write down somewhere that we've decided "associated files" needs to go in the json object for the puzzle, not in the body.